src/bootstrap/config.rs

```diff
@@ -1910,11 +1939,28 @@ impl Config {
 
         // Look for a version to compare to based on the current commit.
         // Only commits merged by bors will have CI artifacts.
+
+        // If we are running in CI, the current commit will never have artifacts already built.
+        let tip_commit = if CiEnv::is_ci() { "HEAD^" } else { "HEAD" };
+        // NOTE: we intentionally don't filter compiler/ and library/ here, even though we ideally
+        // would to avoid unnecessary downloads. The reason for that is complicated:
+        // - When bors creates a merge commit for a rollup, there are two levels of indirection:
+        //   1. The merge by bors ("Auto merge of...")
+        //   2. The merge from each individual commit in the rollup ("Rollup merge of ...")
+        // - Because of the indirection, `git log --author=bors compiler/` *only* includes merges that weren't from a rollup.
+        // - To avoid filtering out rollup merges, we need to do something like
```

Comment:
Please rephrase to indicate whether rollup merges here refers to "auto merge ..." or "rollup of .." commits. (I'd give a suggestion but I'm not actually sure).

Reply:
this comment is gone now - happy to answer questions on https://github.com/rust-lang/rust/issues/113250